checker/src/org/checkerframework/checker/index/samelen/SameLenVisitor.java

```diff
@@ -25,6 +25,23 @@ public SameLenVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    // This variable is used to store the name of the variable being assigned to on
+    // the left hand side of an assignment/pseudo-assignment, so that that variable
+    // can later be included in the SameLen type. This pattern is a bit uncomfortable,
+    // but there is not a clear other way to save this information.
```

Comment:
Convert the VariableTree to an Element.  If the ElementKind is  a LOCAL_VARIABLE, RESOURCE_VARIABLE, EXCEPTION_PARAMETER, or PARAMETER, `getName()` returns the canonical name.  Otherwise, do something like this:

```
Receiver  receiverF = FlowExpressions.internalRepOfImplicitReceiver(ele);
FlowExpressionContext context = new FlowExpressionContext(receiverF, null, factory.getContext());
Receiver result =FlowExpressionParseUtil.parse(tree.getName(), context, getCurrentTreePath(), false);
```

This should probably be added as a new method to FlowExpressions. (`getInternalReprOf(Variable tree,...)`)